Issue996 hydronictest

JavierArroyoBastida opened the pull request:
This is to increase radiator thermal capacity and occupancy schedule

Mathadon commented on IDEAS/Examples/IBPSA/TwoWayOverwriteTRV.mo:
```diff
@@ -0,0 +1,231 @@
+    annotation (Placement(transformation(extent={{40,60},{60,80}}),
+        iconTransformation(extent={{40,60},{60,80}})));
+
+  IBPSA.Utilities.IO.SignalExchange.Overwrite oveVal(description=
```
Please use IDEAS components only in IDEAS.

JavierArroyoBastida replied:
https://github.com/open-ideas/IDEAS/pull/1111/commits/8795163e7adcff1dbef3bd7ef6fa9faf2b9ab412

Mathadon commented on IDEAS/Examples/IBPSA/TwoWayOverwriteTRV.mo:
```diff
@@ -0,0 +1,231 @@
+within IDEAS.Examples.IBPSA;
```
Can you inline this component in `IDEAS/Examples/IBPSA/SingleZoneResidentialHydronic.mo` since it should not be used elsewhere?

JavierArroyoBastida replied:
Sorry I'm not sure what you mean by inline. I think the best would be to create a base classes package under Examples.IBPSA and insert it in there. That highlights the fact that shouldn't be used somewhere else.

Mathadon commented on IDEAS/Examples/IBPSA/TwoWayOverwriteTRV.mo:
```diff
@@ -0,0 +1,231 @@
+to reflect the typical delay of radiator knobs.
+</p>
+</html>",
+revisions="<html>
```
Can you reset revision history and add an entry?

JavierArroyoBastida replied:
https://github.com/open-ideas/IDEAS/pull/1111/commits/2a13df04f1bdf93b47becd3188616a04ce51d081

Mathadon commented on IDEAS/Examples/IBPSA/SingleZoneResidentialHydronic.mo:
```diff
@@ -75,14 +76,16 @@ model SingleZoneResidentialHydronic
     "Block for reading the pump electrical power"
     annotation (Placement(transformation(extent={{60,20},{80,40}})));
-  Fluid.Actuators.Valves.TwoWayTRV val(
+  TwoWayOverwriteTRV               val(
```
see mail whether we really want this

JavierArroyoBastida replied:
I see the point but I think that the possibility should be there to overwrite the actuator signal. The users can then decide whether they assume that there is a thermostatic valve (no overwrite) or that there is a motorized valve (if there is overwrite) to control the radiator.

Mathadon replied:
The point of the BOPTEST is that the _developers_ decide what the allowed control actions are. Otherwise objective comparison between approaches is not possible. Since this is a simple residential house, I'm not inclined to put a motorized valve in it. @dhblum what do you think?

JavierArroyoBastida replied:
Indeed, developers decide. But both setpoints and actuator signals should be accessible to explore different control approaches. Actually, if we limit this emulator to control only the boiler supply temperature there will not be a point of an external controller at all since the valve will close when the boiler supply temperature increases.

Mathadon replied:
That's the whole point: we should provide to the user _emulators_ that represent physical reality. It's for the same reason that I argued before whether or not we should include default RBC-implementations, which I'm not fond of either, but at least the control options are still physical. I have a house that works exactly like this: I can control the supply water temperature (and pump on/off signal, which we could add here too) and nothing else.

Mathadon replied:
@JavierArroyoBastida @dhblum I propose to not allow thermostatic valve set point overrides. Do you agree?

JavierArroyoBastida replied:
I still think that both, set points and actuator signals should be accessible. Actually I think that we should allow:

1. Overwrite of valve temperature set point.
2. Overwrite of valve actuator possition.
3. Overwrite of pump on/off signal (agreeing with what said in https://github.com/open-ideas/IDEAS/pull/1111#discussion_r382524789).

dhblum replied:
Previously, the heater supply temperature was the control variable, which I presumed could be used by a controller to maintain zone temperature in the space.  With the TRV, control of zone temperature is achieved through modifying mass flow instead of supply temperature.  Adding the TRV without making it in any way programmable (at least through zone temperature setpoint adjustments) means we can't really control the system with different strategies.  As Javi pointed out, changing the water heater temperature wouldn't mean much until the TRV is saturated or at 0.  I'm not sure what the value is in such an emulator with very little controllability.  Unless the pump is controllable, in which case, the primary means of controlling zone temperature with alternative control strategies would be turning the pump on/off.  

I think controllability can still be achieved with the original formulation that Filip indicates is representative of houses in Belgium.  In that, the heater supply water temperature can be controlled to maintain house setpoint.  Is this control strategy indeed realistic?  Or is it turning the pump on/off that maintains the zone temperature comfortable?  

If the TRV is more realistic, then it should be assumed programmable in some way.  Even if it is just the temperature setpoint and not the valve position itself.  Honeywell sells TRV's with temperature setpoint programming through wireless communications: https://heatingcontrols.honeywellhome.com/products/thermostats/thermostatic-radiator-valves/Wirelessly-Controlled-TRVs/evohome-Multi-Zone-Kit/

JavierArroyoBastida replied:
@dhblum thanks for commenting on this. We added a TRV to have a baseline controller as TRVs are quite common and the scheme is thus realistic. There is the other scheme that Filip pointed out: control of whater supply temperature + on/off pump control. This is more similar to the original implementation but we were missing the baseline controller. I'm not sure which one is more common, but all the residential houses I'm familiar with in Belgium and Spain work with the latter one (supply water temperature + on/off pump, **without** TRV, they use manual valves that are set to balance heat distribution to the rooms). My suggestion is to keep either one scheme or the other, which means to pick between: 

1. Original scheme: remove TRV and control supply water temperature + on/off pump, but adding a simple and realistic RBC with something like hysteresis or a heating curve.

2. We keep the TRV but we then also allow control of either valve temperature set point, or valve openning, or both. 

I think I prefer the first setting. @Mathadon what do you think?

Mathadon replied:
In practice there is often 1) one room + thermostat + radiator without TRV and 2) multiple rooms with a radiator with TRV. The thermostat will turn off the whole system when the monitored room reaches the set point and the other rooms will be controlled using a TRV. This model has only one zone, so it would make most sense to include 1) and not 2). This is the motivation for the original implementation. I still feel that it makes sense so I agree with Javi's suggestion. @dhblum ok for you?

dhblum replied:
Yes that sounds ok to me as well.  Thank you both.

Mathadon commented on IDEAS/Examples/IBPSA/SingleZoneResidentialHydronic.mo:
```diff
@@ -56,12 +56,13 @@ model SingleZoneResidentialHydronic
-  Modelica.Blocks.Sources.RealExpression yOcc(y=if calTim.hour > 7 and calTim.hour
-         < 20 then 1 else 0)
+  Modelica.Blocks.Sources.RealExpression yOcc(y=if (calTim.hour < 7 or calTim.hour
+         > 19) or calTim.weekDay > 5 then 1 else 0)
```
heating day and night during the weekend?

JavierArroyoBastida replied:
This would rather mean that the occupant has not much social life and spends the entire weekend at home. The heating is continuously working but only because the emission pump input type, boiler supply temperature and TRV setpoints are constant.

JavierArroyoBastida replied:
https://github.com/open-ideas/IDEAS/pull/1111/commits/bd996382ee551772ffb685329a6141b4f9aa7714

Mathadon replied:
thx!

Mathadon commented on IDEAS/Examples/IBPSA/SingleZoneResidentialHydronic.mo:
```diff
@@ -2,6 +2,218 @@ within IDEAS.Examples.IBPSA;
+    annotation (Line(points={{-52,40},{-40,40}}, color={0,0,127}));
+  connect(oveVal.y, val.y)
+    annotation (Line(points={{-17,40},{0,40},{0,12}}, color={0,0,127}));
+annotation (
```
The annotation can be completely removed. Ideally the model is in the protected section (at the end of the model).

JavierArroyoBastida replied:
I've moved the submodel to the protected section in https://github.com/open-ideas/IDEAS/pull/1111/commits/c483343398d29ecd7928df80854a5385fdc5c18e
but kept the annotation to allow graphical inspection of the submodel.

JavierArroyoBastida commented:
@Mathadon and @dhblum I've already removed TRV and implemented thermostat with hysteresis following our discussion above https://github.com/open-ideas/IDEAS/pull/1111#discussion_r386691022. Please let me know what you think.

Mathadon commented on IDEAS/Examples/IBPSA/SingleZoneResidentialHydronic.mo:
```diff
@@ -56,39 +57,272 @@ model SingleZoneResidentialHydronic
+        rotation=180,
+        origin={30,-70})));
+protected
+  model TwoWayOverwriteTRV
```
@JavierArroyoBastida this model can be removed?

JavierArroyoBastida replied:
@Mathadon you're right, that's not needed anymore. Removed in https://github.com/open-ideas/IDEAS/pull/1111/commits/90acc02d5156a1c0d0082ce3187d4afa65b8cee1.
